Source/DafnyCore/Generic/NodeExtensions.cs

```diff
@@ -119,14 +119,13 @@ public static class NodeExtensions {
     }
 
     var newParent = new LList<INode>(node, parent);
+    var result = new List<LList<INode>>();
```

Comment:
On line 118, should it return `new LList<INode>[] {}` ?

Also, would `return node.Children.SelectMany(child => child.FindNodeChainsInUri(uri, newParent))` be more performant?

Reply:
I will do the first two changes.
For the third, I love unit tests and I started to write some myself in DafnyCore, but I don't know exactly what this function does, why it does it, what I should require and ensure, and I don't have the bandwidth to investigate. At least, it's covered by a test in the language server.